Lib/fontTools/otlLib/optimize.py

```diff
@@ -0,0 +1,493 @@
+            if self._column_indices is None:
+                bitmask = reduce(int.__or__, (lines[i] for i in self.indices))
+                self._column_indices = bit_indices(bitmask)
+            return self._column_indices
+
+        @property
+        def width(self):
+            # Width = number of columns that have a 1 in at least 1 line
+            #   => binary OR all the lines and then count bits
+            return len(self.column_indices)
+
+        @property
+        def cost(self):
+            if self._cost is None:
+                # From: https://docs.microsoft.com/en-us/typography/opentype/spec/gpos#pair-adjustment-positioning-format-2-class-pair-adjustment
+                self._cost = (
+                    # uint16	posFormat	Format identifier: format = 2
+                    2
+                    # Offset16	coverageOffset	Offset to Coverage table, from beginning of PairPos subtable.
+                    + 2
+                    + self.coverage_bytes
+                    # uint16	valueFormat1	ValueRecord definition — for the first glyph of the pair (may be zero).
+                    + 2
+                    # uint16	valueFormat2	ValueRecord definition — for the second glyph of the pair (may be zero).
+                    + 2
+                    # Offset16	classDef1Offset	Offset to ClassDef table, from beginning of PairPos subtable — for the first glyph of the pair.
+                    + 2
+                    + self.classDef1_bytes
+                    # Offset16	classDef2Offset	Offset to ClassDef table, from beginning of PairPos subtable — for the second glyph of the pair.
+                    + 2
+                    + self.classDef2_bytes
+                    # uint16	class1Count	Number of classes in classDef1 table — includes Class 0.
+                    + 2
+                    # uint16	class2Count	Number of classes in classDef2 table — includes Class 0.
+                    + 2
+                    # Class1Record	class1Records[class1Count]	Array of Class1 records, ordered by classes in classDef1.
+                    + (valueFormat1_bytes + valueFormat2_bytes)
+                    * len(self.indices)
+                    * self.width
+                )
+            return self._cost
+
+        @property
+        def coverage_bytes(self):
+            # TODO: (Jany) actually figure out which format would be the best?
+            # For now, always Format 1
+            # From https://docs.microsoft.com/en-us/typography/opentype/spec/chapter2#coverage-format-1
+            return (
+                # uint16	coverageFormat	Format identifier — format = 1
+                2
+                # uint16	glyphCount	Number of glyphs in the glyph array
+                + 2
+                # uint16	glyphArray[glyphCount]	Array of glyph IDs — in numerical order
+                + sum(len(all_class1[i]) for i in self.indices) * 2
+            )
+
+        @property
+        def classDef1_bytes(self):
```

Comment:
For classDef1, you can use the class=0 smartly, assigning it to whichever class that saves most bytes. That can be implemented in your optimizer.